src/test/java/org/dependencytrack/ResourceTest.java

```diff
@@ -86,37 +83,21 @@ public abstract class ResourceTest {
     // Hashing is expensive. Do it once and re-use across tests as much as possible.
     protected static final String TEST_USER_PASSWORD_HASH = new String(PasswordService.createHash("testuser".toCharArray()));
 
-    protected QueryManager qm;
     protected Team team;
     protected String apiKey;
 
-    @BeforeClass
+    @BeforeAll
     public static void init() {
         Config.enableUnitTests();
     }
 
-    @Before
-    public void before() throws Exception {
+    @BeforeEach
+    final void initResourceTest() throws Exception {
         // Add a test user and team with API key. Optional if this is used, but its available to all tests.
-        this.qm = new QueryManager();
         team = qm.createTeam("Test Users");
         this.apiKey = qm.createApiKey(team).getKey();
     }
 
-    @After
-    public void after() throws Exception {
-        // PersistenceManager will refuse to close when there's an active transaction
-        // that was neither committed nor rolled back. Unfortunately some areas of the
-        // code base can leave such a broken state behind if they run into unexpected
-        // errors. See: https://github.com/DependencyTrack/dependency-track/issues/2677
-        if (!qm.getPersistenceManager().isClosed()
-                && qm.getPersistenceManager().currentTransaction().isActive()) {
-            qm.getPersistenceManager().currentTransaction().rollback();
-        }
-
-        PersistenceManagerFactory.tearDown();
-    }
```

Comment:
Why would this no longer be necessary?

Reply:
The ResourceTest is now inherited from PersistenceCapableTest, which does the exactly same thing.